### PR TITLE
Improved gcorr iteration script

### LINKED ISSUE #21: Consolidate `gcorr` scripts
The `gcorr` computation is currently distributed among a series of scripts, and written in a way that requires a cluster for submitting jobs.  This code should be consolidated into one function, similar to `xfaster_run()`, which can run as a single multiprocessing job with a series of relevant checkpoints.

### PR DESCRIPTION
**Pull request by sGourapura**
Refactor gcorr script to work locally or submitted online, iterate until convergence, and condense outputs into a single directory with easily identifiable intermediate iteration files.

Closes #21

### CODE REVIEW

**Review by arahlin — CHANGES_REQUESTED** [review]
This script is just the same as iterate.py with small changes to the way run_xfaster is called?  If so, just make the changes as an option in the original iterate.py script, otherwise it's hard to follow what you've actually changed.

File: scripts/gcorr/iterate.py [hunk 1]
```diff
@@ -173,40 +180,73 @@
...
-    g=args.gcorr_config, o=run_name_iter
-)
+if args.no_submit:
+    cmd = "python run_xfaster.py --gcorr-config {g} --omp 1 --check-point transfer --reload-gcorr --no-submit -o {o} > /dev/null".format(
+        g=args.gcorr_config, o=run_name_iter
+    )
+else:
+    cmd = "python run_xfaster.py --gcorr-config {g} --omp 1 --check-point transfer --reload-gcorr -o {o} > /dev/null".format(
+        g=args.gcorr_config, o=run_name_iter
+    )
```
Comment (arahlin) [inline scripts/gcorr/iterate.py:190]:
```suggestion
cmd = "python run_xfaster.py --gcorr-config {g} --omp 1 --check-point transfer --reload-gcorr {s} -o {o} > /dev/null".format(
    g=args.gcorr_config, o=run_name_iter, s="--no-submit" if args.no_submit else ""
)
```

File: scripts/gcorr/iterate.py [hunk 2]
```diff
@@ -173,40 +180,73 @@
...
-    g=args.gcorr_config, o=run_name_iter, n=int(g_cfg["gcorr_opts"]["nsim"]) - 1
-)
+if args.no_submit:
+    cmd = "python run_xfaster.py --gcorr-config {g} --omp 1 --check-point bandpowers -o {o} -f 1 -n {n} --no-submit > /dev/null".format(
+        g=args.gcorr_config, o=run_name_iter, n=int(g_cfg["gcorr_opts"]["nsim"]) - 1
+    )
+else:
+    cmd = "python run_xfaster.py --gcorr-config {g} --omp 1 --check-point bandpowers -o {o} -f 1 -n {n} > /dev/null".format(
+        g=args.gcorr_config, o=run_name_iter, n=int(g_cfg["gcorr_opts"]["nsim"]) - 1
+    )
```
Comment (arahlin) [inline scripts/gcorr/iterate.py:236]:
```suggestion
cmd = "python run_xfaster.py --gcorr-config {g} --omp 1 --check-point bandpowers -o {o} -f 1 -n {n} {s} > /dev/null".format(
    g=args.gcorr_config, o=run_name_iter, n=int(g_cfg["gcorr_opts"]["nsim"]) - 1, s="--no-submit" if args.no_submit else ""
)
```

File: scripts/gcorr/iterate.py [hunk 3]
```diff
@@ -173,40 +180,73 @@
...
 os.system(cmd)
 
 transfer_exists = {}
 for tag in tags:
     transfer_exists[tag] = False
-while not np.all(list(transfer_exists.values())):
-    # wait until transfer functions are done to submit rest of jobs
+
+# If running in series, check that transfer function exists. Ff not, error and exit
+if args.no_submit:
```
Comment (arahlin) [inline scripts/gcorr/iterate.py:199]: Minimize diff
```suggestion
while not np.all(list(transfer_exists.values())):
    # wait until transfer functions are done to submit rest of jobs
```

File: scripts/gcorr/iterate.py [hunk 4]
```diff
@@ -173,40 +180,73 @@
...
+        for tag in tags:
+            rundirf = os.path.join(rundir, tag)
+            transfer_files = glob.glob(
+                os.path.join(rundirf, "transfer_all*{}.npz".format(tag))
+            )
+            transfer_exists[tag] = bool(len(transfer_files))
+
+        print("transfer exists: ", transfer_exists)
+        time.sleep(15)
+
```
Comment (arahlin) [inline scripts/gcorr/iterate.py:225]:
```suggestion
    print("transfer exists: ", transfer_exists)
    if args.no_submit:
        if np.all(list(transfer_exists.values())):
            break
        raise RuntimeError("Some/all transfer functions not made: {}".format(transfer_exists))
    else:
        time.sleep(15)
```

**Review by arahlin — CHANGES_REQUESTED** [review]
A couple of diff-minimizing code change suggestions.  You can batch-commit them from the web interface by going to the "Files Changed" tab on this PR and clicking on "Add suggestion to batch" for each one that you're ok with, then committing all of them at once.

**Review by sGourapura — COMMENTED** [review]
Works for me. Neat this can be done in the gitweb

**Comment by sGourapura** [comment]
Is there no pushing because we are editing the remote copy directly?

**Comment by arahlin** [comment]
> Is there no pushing because we are editing the remote copy directly?

That's right.  You can pull these changes in to your checked out copy on your computer with a regular `git pull`.

**Review by arahlin — APPROVED** [review]
